src/Analysers/LikelihoodInputAnalyser.cpp

```diff
@@ -15,34 +15,198 @@ void LikelihoodInputAnalyser::analyse(const EventPtr event) {
 
 	weight_ = event->weight();
 	treeMan_->setCurrentFolder(histogramFolder_);
+	unsigned int parton = 0;
+	double CSV;
+
+	const JetCollection AllJets(event->Jets());
+	const JetCollection jets(event->CleanedJets());
+	const JetCollection bjets(event->CleanedBJets());
+	// const JetCollection genJets( event->GenJets() );
+
+	METAlgorithm::value metType = (METAlgorithm::value) 0;
+	const METPointer met(event->MET(metType));
+
+	if (event->isSemiLeptonicElectron() || event->isSemiLeptonicMuon()){
+		for ( unsigned int AllJetIndex=0; AllJetIndex < AllJets.size(); ++AllJetIndex ) {
+
+			JetPointer AllJet = AllJets[AllJetIndex];
+
+			if (AllJet->matched_parton() == 0) continue;//if null pointer
+
+			parton = abs(AllJet->matched_parton()->pdgId());
+			CSV = AllJet->getBTagDiscriminator(BAT::BtagAlgorithm::value::CombinedSecondaryVertexV2);
+
+
+			// Look at matched generator jets <----------- something im working on (lookiong at differences between jet pt and gen jet pt etc)
+			// const ParticlePointer genJet = AllJet->matched_generated_jet();
+			// if ( genJet != 0 ) {
+			// 	treeMan_->Fill("genJetPt",genJet->pt());
+			// 	treeMan_->Fill("genJetEta",genJet->eta());
+			// 	treeMan_->Fill("JetPt", AllJet->pt());
+			// 	treeMan_->Fill("JetEta", AllJet->eta());
+			// }
+
+
+			if ( abs(CSV) >= 11 ){
+				cout << "CSV : " << CSV << " Run Number : " << event->runnumber() << " Event Number : " << event->eventnumber() << " Lumi Number : " << event->lumiblock() << endl;
+				// event->inspect(); //For more info on events failing
+				continue;
+				}
+
+			if ( (parton >= 1 && parton <= 4) || (parton == 21) ){
+				treeMan_->Fill("LightJets", CSV);
+				// cout << "LightJet CSV : " << CSV << endl;
+				if ( CSV == -10 ){
+					treeMan_->Fill("LightJetPt", AllJet->getFourVector().Pt());
+					treeMan_->Fill("LightJetEta", AllJet->getFourVector().Eta());
```

Comment:
Remove these plots for now?  A bit misleading if you fill a branch called "LightJetPt" and only jets that have weird CSV are considered.